src/hnsw/htab_cache.c

```diff
@@ -0,0 +1,53 @@
+#include "htab_cache.h"
+
+HTABCache cache_create(char *name, MemoryContext ctx)
+{
+    HTABCache cache;
+
+    HASHCTL hctl;
+    hctl.keysize = sizeof(HTABCacheKey);
+    hctl.entrysize = sizeof(HTABCacheEntry);
```

Comment:
can you add an assertion here that the passed memory context is not the global (I think it is called `TopMemoryContext`). We risk memory leaks if we allocate there.